Cesium.Parser/TokenExtensions.cs

```diff
@@ -11,7 +13,34 @@ public static string UnwrapStringLiteral(this IToken<CTokenType> token)
         if (token.Kind != CTokenType.StringLiteral)
             throw new ParseException($"Non-string literal token: {token.Kind} {token.Text}");
 
-        // TODO[#235]: More thorough unwrap for more literal types.
-        return token.Text.Trim('"').Replace("\\n", "\n");
+        // simple escape sequences
+        var result = token.Text.Trim('"')
+            .Replace("\\'", "\'")
+            .Replace("\\\"", "\"")
+            .Replace("\\?", "?")
+            .Replace("\\\\", "\\")
+            .Replace("\\a", "\a")
+            .Replace("\\b", "\b")
+            .Replace("\\f", "\f")
+            .Replace("\\n", "\n")
+            .Replace("\\r", "\r")
+            .Replace("\\t", "\t")
+            .Replace("\\v", "\v");
+
+        // numeric escape sequences
+        result = Regex.Replace(result, @"\\([0-7]{1,3})", m =>
+            char.ConvertFromUtf32(Convert.ToInt32(m.Groups[1].Value, 8)));
+
+        result = Regex.Replace(result, @"\\[xX]([0-9a-fA-F]{2})", m =>
+            char.ConvertFromUtf32(Convert.ToInt32("0x" + m.Groups[1].Value, 16)));
+
+        //universal character names
+        result = Regex.Replace(result, @"\\u([0-9a-fA-F]{4})", m =>
+            char.ConvertFromUtf32(Convert.ToInt32("0x" + m.Groups[1].Value, 16)));
+
+        result = Regex.Replace(result, @"\\U([0-9a-fA-F]{8})", m =>
+            char.ConvertFromUtf32(Convert.ToInt32("0x" + m.Groups[1].Value, 16)));
```

Comment:
The general approach looks pretty okay, but this code creates a lot of temporary strings. Could we somehow improve that?

Say, for chained `Replace` you could use a `StringBuilder`, but then again: it is not ultra efficient and may be done better (with a single pass over the initial string).

When you develop a C compiler, it's sometimes hard to decide on what level of optimization you should stop.

Reply:
I’m fine with original approach. If @BaLiKfromUA want make this more efficient we can allocate `StringBuilder` with capacity same as original string and walk string in one pass replacing characters as needed. If that’s too complicated let us know. 
also make tests pass. Comment problematic cases as needed, but please do not remove.

Reply:
I can implement it without any problems I guess. Initially, I just followed the original style + was trying to solve the task first.

Reply:
If you would like to join Russian speaking community you can go https://t.me/cilchat where we hanging out. If not, we can obviously communicate here.

Reply:
I pushed my interpretation of your idea, hope it fits

Reply:
Yeah, it very much does.